src/Pulsar.Client/Api/Producer.fs

```diff
@@ -14,11 +14,14 @@ open System.Collections.Generic
 type ProducerException(message) =
     inherit Exception(message)
 
+type private PendingMessage =
+    { SequenceId: SequenceId
+      Payload: Payload }
+
 type Producer private (producerConfig: ProducerConfiguration, lookup: BinaryLookupService) as this =
     let producerId = Generators.getNextProducerId()
     let messages = ConcurrentDictionary<SequenceId, TaskCompletionSource<MessageId>>()
```

Comment:
Can you try refactoring messages concurrent dictionary to pendingCallbacks queue like in Java version? Or maybe join two into one queue with tuple argument?